cpp/src/arrow/compute/exec.h

```diff
@@ -174,6 +174,10 @@ class ARROW_EXPORT SelectionVector {
 /// TODO: Datum uses arrow/util/variant.h which may be a bit heavier-weight
 /// than is desirable for this class. Microbenchmarks would help determine for
 /// sure. See ARROW-8928.
+
+/// \addtogroup execnode-components
```

Comment:
This looks like a new group.  You need to add a defgroup [here](https://github.com/apache/arrow/blob/master/cpp/src/arrow/compute/api.h) if you want this to work correctly.

Are you able to build these changes locally to test?

Reply:
This builds perfectly fine. And displays right. I didn’t add it because I assumed it followed by the header file import. When the corresponding headers included the build breaks. I didn’t investigate deeper yet. But removing the group and header import was still fine. 

I guess we could just include the group only, without the header?